cromwell-backend/src/main/scala/cromwell/backend/config/BackendConfiguration.scala

```diff
@@ -0,0 +1,46 @@
+package cromwell.backend.config
+
+import com.typesafe.config.ConfigFactory
+
+import scala.collection.JavaConversions._
+
+/**
+  * Defines a backend configuration.
+  * @param name Backend name.
+  * @param initClass Initialization class for the specific backend.
+  */
+case class BackendConfigurationEntry(name: String, initClass: String)
+
+object BackendConfiguration {
+  val config = ConfigFactory.load()
+  val backendCfg = config.getConfig("backend")
+  val defaultBackend = backendCfg.getString("default")
+  val backendProviders = backendCfg.getConfigList("providers")
+  val backendList = backendProviders.map(entry =>
+    BackendConfigurationEntry(entry.getString("name"), entry.getString("initClass"))).toList
+
+  def apply(): BackendConfiguration = new BackendConfiguration(backendList, defaultBackend)
+}
+
+/**
+  * Retrieves backend configuration.
+  * @param backendList List of backend entries in configuration file.
+  * @param defaultBackend Backend name to be used as default.
+  */
+class BackendConfiguration(backendList: List[BackendConfigurationEntry], defaultBackend: String) {
```

Comment:
I disagree @kshakir - I find that the whole "use the most generic interface possible" is a bit of a holdover from java thinking, one which I don't see the point for. How many bugs have we pointed out where people are passing around a Seq but using the cons operator? How can I reason about the runtime complexity of a function if I don't know what kind of seq is being passed around? Also note that there is no notion of mutable.Seq vs immutable.Seq.

For a long while I was on Vector to mean "bag of stuff" but now I'm fully supportive of List.